routes/authentication_cr.js

```diff
@@ -0,0 +1,81 @@
+const express = require("express");
+const router = express.Router()
+const bcrypt = require("bcryptjs")
+
+//importing local files
+const db = require("../db/models");
+const { environment } = require("../config");
+const { asyncHandler, csrfProtection, signUpValidator, validationResult, loginValidators } = require("./utiles");
+
+//Sing-Up
+
+router.post("/sign-up", csrfProtection, signUpValidator, asyncHandler(async(req, res) => {
+
+    const { username, email, password } = req.body
+
+    const validatorError = validationResult(req)
+    if(validatorError.isEmpty()) {
+        const hashedPassword = await bcrypt.hash(password, 10);
+        await db.User.create({
+            username,
+            email,
+            password: hashedPassword,
+            createdAt: new Date(),
+            updatedAt: new Date()
+        });
+
+
+        res.json({}); // Authentication Token to be inserted in the future
+
+
+    } else {
+        const errors = validatorError.array().map((error) => error.msg);
+        res.json({ errors });
+    }
+}))
+
+router.get("/demo", asyncHandler( async (req, res) => {
+    const user = await db.User.findOne({
+        where: {
+            username: "DemoUser",
+            email: "DemoUser@NatlRoutes.com"
+        }
+    });
+    console.log(user);
+    if (!user) {
+        res.send("Please seed all files");
+    } else {
+        req.session.auth = {userId: user.id, username: user.username};
```

Comment:
You may want to use `req.session.save` to avoid a race condition with your redirect. Something like this:

```js
req.session.auth = {userId: user.id, username: user.username};
req.session.save(err => {
  if (err) return next(err);
  // at this point, you have guaranteed your session information is stored and ready to be used for the next request
  res.redirect("/my-routes");
})
```

Reply:
Oh, sure! Completely forgot about this part! Thank you for reminding me!